### ISSUE TITLE
Keras Model cannot be saved as .h5 when tf.keras.backend.clip is used

### ISSUE THREAD

**Issue by oholimoli** [tensorflow/tensorflow#33565]
**System information**
- Tensorflow 2.0.0
- Python 3.7.4

**Current behavior**
I'm using the `tf.keras.backend.clip` function in a model and cannot save it because of a naming problem: Two layers have the same name (see `tf_op_layer_clib_by_value_16` in the ModelToDot output below). This causes a problem when exporting it to hdf5.

Couldn't find a workaround or way to rename one of the two layers.

**Code to reproduce the issue**
```
import tensorflow as tf
from tensorflow import keras

input1 = keras.layers.Input(shape = (16,), name = iname)
dense1 = keras.layers.Dense(1, name = 'dense1', activation = None, # relu
                       #kernel_initializer=keras.initializers.constant(dAB1_kernel),
                       #kernel_constraint=keras.constraints.NonNeg(),
                       use_bias=True)(input1)
dense1_clip = tf.keras.backend.clip(dense1,min_value = 0, max_value = 1000)

model1 = keras.Model(input1, dense1_clip)

```

![bug](https://user-images.githubusercontent.com/47213738/67195858-0606e700-f3fa-11e9-8ad4-80d6f425fe63.PNG)

**Other info / logs**
```
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-179-97bae6c176be> in <module>
      7 
      8 model1 = keras.Model(input1, dense1_clip)
----> 9 model1.save('m1.h5')

d:\programme\python37\lib\site-packages\tensorflow_core\python\keras\engine\network.py in save(self, filepath, overwrite, include_optimizer, save_format, signatures, options)
    973     """
    974     saving.save_model(self, filepath, overwrite, include_optimizer, save_format,
--> 975                       signatures, options)
    976 
    977   def save_weights(self, filepath, overwrite=True, save_format=None):

d:\programme\python37\lib\site-packages\tensorflow_core\python\keras\saving\save.py in save_model(model, filepath, overwrite, include_optimizer, save_format, signatures, options)
    110           'or using `save_weights`.')
    111     hdf5_format.save_model_to_hdf5(
--> 112         model, filepath, overwrite, include_optimizer)
    113   else:
    114     saved_model_save.save(model, filepath, overwrite, include_optimizer,

d:\programme\python37\lib\site-packages\tensorflow_core\python\keras\saving\hdf5_format.py in save_model_to_hdf5(model, filepath, overwrite, include_optimizer)
    107     model_weights_group = f.create_group('model_weights')
    108     model_layers = model.layers
--> 109     save_weights_to_hdf5_group(model_weights_group, model_layers)
    110 
    111     # TODO(b/128683857): Add integration tests between tf.keras and external

d:\programme\python37\lib\site-packages\tensorflow_core\python\keras\saving\hdf5_format.py in save_weights_to_hdf5_group(f, layers)
    623 
    624   for layer in layers:
--> 625     g = f.create_group(layer.name)
    626     weights = _legacy_weights(layer)
    627     weight_values = K.batch_get_value(weights)

d:\programme\python37\lib\site-packages\h5py\_hl\group.py in create_group(self, name, track_order)
     66             name, lcpl = self._e(name, lcpl=True)
     67             gcpl = Group._gcpl_crt_order if track_order else None
---> 68             gid = h5g.create(self.id, name, lcpl=lcpl, gcpl=gcpl)
     69             return Group(gid)
     70 

h5py\_objects.pyx in h5py._objects.with_phil.wrapper()

h5py\_objects.pyx in h5py._objects.with_phil.wrapper()

h5py\h5g.pyx in h5py.h5g.create()

ValueError: Unable to create group (name already exists)
```

**Comment by gowthamkpr** [comment]
@oholimoli Tried to reproduce your issue in google colab using the lastest version of Tensorflow i.e., 2.1.0-dev20191021 but was not able to reproduce it. Please find the gist [here](https://colab.sandbox.google.com/gist/gowthamkpr/5cf063d9e3509684b288f1839b1252e4/untitled203.ipynb#scrollTo=b9F2THOpllp3).

**Comment by oholimoli** [comment]
If you try model.save('model.h5') instead you will get the same error.

**Comment by gowthamkpr** [comment]
@oholimoli This is indeed a bug in Tensorflow 2.0. In tensorflow 2.0, you can change ".h5" to ".tf" and everything should be saved.

There is also a workaround to save the model in.h5 format and its mentioned in this [comment](https://github.com/keras-team/keras/issues/12195#issuecomment-544416524).

**Comment by ymodak** [comment]
This is fixed with latest tf nightly '2.2.0-dev20200129'. Thanks!
